lib/active_record/connection_adapters/trilogy_adapter.rb

```diff
@@ -70,13 +70,35 @@ def last_inserted_id(result)
 
       include DatabaseStatements
 
+      SSL_MODES = {
+        SSL_MODE_DISABLED: Trilogy::SSL_DISABLED,
+        SSL_MODE_PREFERRED: Trilogy::SSL_PREFERRED_NOVERIFY,
+        SSL_MODE_REQUIRED: Trilogy::SSL_REQUIRED_NOVERIFY,
+        SSL_MODE_VERIFY_CA: Trilogy::SSL_VERIFY_CA,
+        SSL_MODE_VERIFY_IDENTITY: Trilogy::SSL_VERIFY_IDENTITY
+      }.freeze
+
       class << self
         def new_client(config)
+          config[:ssl_mode] = parse_ssl_mode(config[:ssl_mode]) if config[:ssl_mode]
           ::Trilogy.new(config)
         rescue Trilogy::DatabaseError => error
           raise translate_connect_error(config, error)
         end
 
+        def parse_ssl_mode(mode)
+          return unless mode
+
+          # return mode if it's already a Trilogy::SSL_MODE
```

Comment:
This comment is a tiny bit confusing to me, since I don't see `Trilogy::SSL_MODE` defined anywhere, but the code that follows makes sense to me even without the comment.

Reply:
Good point! Honestly, Copilot wrote that comment and I misread it and thought it it said `SSL_MODES`, which is the hash with all the ssl mode values introduced in this PR. I can remove it though. 😃

Reply:
```suggestion
```